app/Services/Project/ProjectAvatarService.php

```
@@ -134,8 +135,21 @@ private function generateS3(string $projectRef, string $projectName): bool
-            // Then upload using Storage:put()
-            Storage::disk('project_thumb')->put($projectRef . '/' . $this->filename, (string) $imageThumbEncoded);
+            // Then upload using Storage::put() with retries
+            $maxRetries = 3;
+            $retryDelay = 1; // seconds
+
+            for ($retry = 0; $retry <= $maxRetries; $retry++) {
+                try {
+                    Storage::disk('project_thumb')->put($projectRef . '/' . $this->filename, (string) $imageThumbEncoded);
+                    break; // Success, exit retry loop
+                } catch (Throwable $e) {
+                    if ($retry === $maxRetries || !($e instanceof S3Exception && Common::isRetryableError($e))) {
```

PHPMD (2.15.0)

Avoid using static access to class '\ec5\Libraries\Utilities\Common' in method 'generateS3'. (Clean Code Rules)

(StaticAccess)